tools/src/wyvern/tools/typedAST/core/declarations/ModuleDeclaration.java

```diff
@@ -51,6 +58,20 @@ public ModuleDeclaration(String name, List imports, List<GenericParameter> gener
         this.args = args;
         this.imports = imports;
         this.generics = generics;
+        this.isAnnotated = isAnnotated;
+        this.effectSet = EffectSet.parseEffects(name, effects, false, location);
+        if (args.isEmpty() && imports.isEmpty() && this.effectSet != null
+                && !this.effectSet.getEffects().isEmpty()) {
+            ToolError.reportError(ErrorMessage.PURE_MODULE_ANNOTATION, FileLocation.UNKNOWN);
```

Comment:
Never use `FileLocation.UNKNOWN` if you can help it.  Here you should use `location`